flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/inference/TypeStrategiesTestBase.java

```diff
@@ -42,26 +40,36 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Base class for tests of {@link TypeStrategies}. */
-@RunWith(Parameterized.class)
 public abstract class TypeStrategiesTestBase {
 
-    @Parameter public TestSpec testSpec;
-
-    @Test
-    public void testTypeStrategy() {
+    @ParameterizedTest(name = "{index}: {0}")
+    @MethodSource(
+            value = {
+                "org.apache.flink.table.types.inference.strategies.ArrayTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.CurrentWatermarkTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.DecimalTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.GetTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.MapTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.MappingTypeStrategiesTest#testData",
+                "org.apache.flink.table.types.inference.strategies.RowTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.StringConcatTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.TypeStrategiesTest#testData"
+            })
```

Comment:
It leads to cartesian of tests and input....
need to think how to cope with this